integration-libs/opf/cta/core/services/opf-dynamic-cta.service.ts

```diff
@@ -0,0 +1,251 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { inject, Injectable } from '@angular/core';
+import {
+  ActiveCartFacade,
+  Cart,
+  CartAddEntrySuccessEvent,
+  CartRemoveEntrySuccessEvent,
+  CartUpdateEntrySuccessEvent,
+} from '@spartacus/cart/base/root';
+import {
+  CurrencyService,
+  EventService,
+  LanguageService,
+  Product,
+  WindowRef,
+} from '@spartacus/core';
+import {
+  CtaEvent,
+  CtaProductItem,
+  CtaScriptsLocation,
+  CtaScriptsRequest,
+  OpfCtaFacade,
+} from '@spartacus/opf/cta/root';
+import { OpfGlobalFunctionsFacade } from '@spartacus/opf/global-functions/root';
+import {
+  GlobalFunctionsDomain,
+  KeyValuePair,
+} from '@spartacus/opf/payment/root';
+import { CurrentProductService } from '@spartacus/storefront';
+import {
+  combineLatest,
+  map,
+  merge,
+  Observable,
+  Subscription,
+  take,
+  tap,
+} from 'rxjs';
+
+@Injectable()
+export class OpfDynamicCtaService {
```

Comment:
[MINOR]
It's not clear at the first sight why we have 3 different services:
- `CtaService`
- `CtaDynamicService`
- `CtaStaticService`

Let's add JSDocs for them, explaining their purpose, relations and differences